proxy/src/transparency/h1.rs

```diff
@@ -81,4 +82,16 @@ pub fn strip_connection_headers(headers: &mut http::HeaderMap) {
     }
 }
 
+/// Checks requests to determine if they want to perform an HTTP upgrade.
+pub fn wants_upgrade<B>(req: &http::Request<B>) -> bool {
+    // HTTP upgrades were added in 1.1, not 1.0.
+    req.version() == http::Version::HTTP_11
+        && req.headers().contains_key(UPGRADE)
```

Comment:
@olix0r noted that we should probably not pass on h2 upgrades, as that would result in the client having an opaque-to-the-proxy h2 connection (we'd just see it as a TCP proxy), and a user couldn't see any normal HTTP telemetry about it. Eventually, we can see those upgrades and upgrade the connection with proxy directly.

- [ ] Check the `Upgrade` header for evidence of an HTTP2 upgrade.
- [ ] Add an explicit transparency test stripping the HTTP2 upgrade.